src/event.cairo

```diff
@@ -84,40 +102,58 @@ mod StarkPassEvent {
             self.ticket_price.read()
         }
 
-        // ##### ERC20 ACTIVE FUNCTIONS #####
+        fn change_organizer(ref self: ContractState) {
+            // Only the current organizer can assign a new organizer
+            self.only_owner();
+
+            // Change the owner
+            let new_owner = get_caller_address();
+            self.organizer.write(new_owner);
+
+            // Emit OwnershipTransfer event
+            self
+                .emit(
+                    Event::OwnershipTransfer(
+                        OwnershipTransfer {
+                            prev_owner: self.organizer.read(), new_owner: new_owner
+                        }
+                    )
+                );
+        }
 
         fn transfer_balance_to_organizer(ref self: ContractState) {
+            // Only the current owner can withdraw the contract balance
+            self.only_owner();
             let organizer = self.organizer.read();
-            assert(get_caller_address() == organizer, 'CALLER_IS_NOT_ORGANIZER');
             let amount = self.get_balance_of_contract();
-            self.transfer(organizer, amount);
-        }
 
-        fn mock_buy_ticket(ref self: ContractState) {
-            let sender = get_caller_address();
-            self.attendees.write(sender, true);
+            // Withdraw the contract's balance to the organizer's address
+            self.transfer(organizer, amount);
         }
 
         fn buy_ticket(ref self: ContractState) {
+            let ticket_recipient = get_caller_address();
             let ticket_price = self.ticket_price.read();
-            let sender = get_caller_address();
 
-            self.create_erc20_dispatcher().transferFrom(
-                get_caller_address(),
-                get_contract_address(),
-                ticket_price
-            );
-            self.attendees.write(sender, true);
+            // Pay in ERC20 if the ticket price is positive i.e., if the event isn't free
+            if ticket_price > 0 {
+                self
```

Comment:
I think we can call the private method `transfer_from`
```
self.transfer_from(ticket_recipient, get_contract_address(), ticket_price);
```

Or, if we're not gonna use it, we can just delete it

Reply:
Great point. Done here and in the other places as well.